src/lib/types/experimental.ts

```diff
@@ -0,0 +1,31 @@
+export interface IFlags {
+    [key: string]: boolean;
+}
+export interface IExperimentalOptions {
+    flags?: {
+        [key: string]: boolean;
+    };
+    anonymiseEventLog?: boolean;
+    userGroups?: boolean;
+    embedProxy?: boolean;
+    batchMetrics?: boolean;
+    dynamicFlags?: string[];
+    externalResolver?: IExternalFlagResolver;
+}
+
+export interface IUIFlags extends IFlags {
+    ENABLE_DARK_MODE_SUPPORT?: boolean;
+}
+
+export interface IFlagContext {
+    [key: string]: string;
+}
+
+export interface IFlagResolver {
+    getUIFlags: (context?: IFlagContext) => IUIFlags;
+    isExperimentEnabled: (flagName: string, context?: IFlagContext) => boolean;
```

Comment:
Not sure if we'll need an `isReady` method to wait for at startup, or events that trigger on flag update. 

Maybe something for later.

Reply:
you could achieve that by deferring the initialization of unleash until the third party flag resolver is ready.